includes/tracks/events/class-wc-coupons-tracking.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * WooCommerce Coupons Tracking
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds actions to track usage of WooCommerce Orders.
+ */
+class WC_Coupons_Tracking {
+	/**
+	 * Init tracking.
+	 */
+	public function init() {
+		add_action( 'load-edit.php', array( $this, 'tracks_coupons_events' ), 10 );
+	}
+
+	/**
+	 * Add a listener on the "Apply" button to track bulk actions.
+	 */
+	public function tracks_coupons_bulk_actions() {
+		wc_enqueue_js(
+			"
+			var apply = $( '#posts-filter input[type=\"submit\"]' );
```

Comment:
This will select all the submits in the page including "Filter" and "Search coupons", but I believe we only want to attach an event listener to "Apply" to track bulk actions. If I'm not missing something, should we maybe then only select the submit elements that we want to monitor? I haven't checked if that is easy/possible.

Reply:
Thanks for seeing this! I see that the target "Apply" buttons have ids on them, so in https://github.com/woocommerce/woocommerce/pull/23189/commits/bc233409096c2693d92892a6d216216844b4d2b9 I just used `#doaction` and `#doaction2`.